### PR TITLE
transition existing vue app to new modernized gauge API

### PR DESCRIPTION
**Pull request by ngottlieb**
This PR will hold most of the changes for #585, but not the edit flows interface (since that can be released separately, later, if needed).

Currently, it includes:

- the Vue aspects of #587 
- beta box changes for #588

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -25,6 +25,8 @@
     "@mapbox/mapbox-gl-draw": "^1.1.2",
     "@mapbox/mapbox-gl-draw-static-mode": "^1.0.1",
     "@tinymce/tinymce-vue": "^3.2.8",
+    "@trpc/client": "^11.0.0-next-beta.264",
+    "@trpc/server": "^11.0.0-next-beta.264",
```
Comment (tallowen) [inline package.json:29]: Does the vue app need to depend on the server?

File: package.json [hunk 2]
```diff
@@ -25,6 +25,8 @@
     "@mapbox/mapbox-gl-draw": "^1.1.2",
     "@mapbox/mapbox-gl-draw-static-mode": "^1.0.1",
     "@tinymce/tinymce-vue": "^3.2.8",
+    "@trpc/client": "^11.0.0-next-beta.264",
```
Comment (tallowen) [inline package.json:28]: I recall having issues with diverging versions. Right now, everything else is on `10.43.0` should we standardize?

File: src/app/views/river-detail/components/main-tab/components/beta-box/beta-box.vue [hunk 3]
```diff
@@ -193,18 +139,28 @@ export default {
...
     ...mapState({
       loading: state => state.RiverDetail.loading,
       river: state => state.RiverDetail.data,
       editMode: state => state.Global.editMode,
-      gages: state => state.RiverGages.data?.gauges ?? [],
-      metrics: state => state.RiverGages.data?.metrics ?? []
     }),
-    readingIsEstimated () {
-      return this.river?.readingsummary?.gauge_estimated
+    // TODO: decide how to handle secondary gauges. do we fall back if primary isn't reporting?
```
Comment (tallowen) [inline src/app/views/river-detail/components/main-tab/components/beta-box/beta-box.vue:151]: cc @AWKevin - I think we decided that not falling back is ultimately more clear.

File: package.json (hunk 1 above)
Comment (ngottlieb) [reply package.json:29]: yeah, unfortunately you can't just use the client

File: package.json (hunk 2 above)
Comment (ngottlieb) [reply package.json:28]: I'll drop back to 10.43 and report back if I have any issues!

File: src/app/views/river-detail/components/main-tab/components/beta-box/beta-box.vue [hunk 4]
```diff
@@ -193,18 +139,28 @@ export default {
...
+    gaugeCorrelation () {
+      if (this.reachDetail?.detail?.correlations) {
+        return this.reachDetail.detail.correlations.find(x => x.isPrimary);
+      }
+      return null;
+    },
+    adjustedReachGrade () {
+      if (this.gaugeCorrelation && this.gaugeCorrelation.status && this.gaugeCorrelation.correlationDetails) {
+        const adjustedGradeKey = this.gaugeCorrelation.status.status.replace(/-([a-z])/g, g => g[1].toUpperCase());
```
Comment (tallowen) [inline src/app/views/river-detail/components/main-tab/components/beta-box/beta-box.vue:160]: We can probably render something different serverside if that's helpful?

**Review by tallowen — COMMENTED** [review]
Looks great! What's our plan for landing this - will does it need anything else before it can go live?

**Comment by ngottlieb** [comment]
@tallowen I guess broadly speaking, I was imagining I would finish the read components of https://github.com/orgs/AmericanWhitewater/projects/4?pane=issue&itemId=57895707 (i.e. everything except the edit flows interface) before we launched it. And that we would take some steps to establish that we were, I guess, ready to be in "production mode" with the new gauge API.

but maybe I'm overthinking and we should just go for it? not a lot to lose...

**Comment by tallowen** [comment]
Makes sense! I think there are a couple things to get done on the backend before general availability but I'd love to start trying the read versions on beta or something.

**Pull request merged**